include/fastdds/dds/subscriber/DataReader.hpp

```diff
@@ -995,6 +999,17 @@ class DataReader : public DomainEntity
             const void* data,
             const SampleInfo* info) const;
 
+    /**
+     * Get the list of locators on which this DataReader is listening.
+     *
+     * @param [out] locators  LocatorList_t where the list of locators will be stored.
+     *
+     * @return NOT_ENABLED if the reader has not been enabled.
+     * @return OK if a list of locators is returned.
+     */
+    RTPS_DllAPI ReturnCode_t get_listening_locators(
+            eprosima::fastrtps::rtps::LocatorList_t& locators) const;
```

Comment:
Which namespace would be the correct one? We already have some classes on `eprosima::fastdds::rtps`, so I would prefer to add them there.

Where do you think is a good place to add the aliases?
Should we add them directly on `Locator.h`?
Is it better to create a separate file with all the `fastdds::rtps` -> `fastrtps::rtps` aliases?

In the future (i.e. v3.0), we should get rid of the `fastrtps` namespace, so having a single place where we have all the necessary aliases may be better.

Reply:
I have removed the forward declarations on b5f45fa

Reply:
My bad on the comment. I really meant change it to ``eprosima::fastdds::rtps``.

I think we can leave ``eprosima::fastdds::rtps`` to keep classes that are specific to RTPS layer, like locators (specific to RTPS in the sense that they deal with the underlying connection channel). I believe other classes in this namespace were put there following this rationale.

I would go for the ``fastdds::rtps`` -> ``fastrtps::rtps`` aliases, since the latter is the one to be deprecated. This is what we've been doing so far, with the QosPolicies, for example. Besides, deprecating the ``fastrtps::rtps`` will be as easy as removing the includes in ``fastrtps`` folder.

Regarding the place for the alias, right now we have:
* ``include/fastdds/rtps/common/Locator.h`` that defines the locator in ``fastrtps::rtps``
* ``include/fastrtps/rtps/common/Locator.h`` which only includes the former, without changing the namespace.
What I think we should do here is:
* Change the namespace in ``include/fastdds/rtps/common/Locator.h`` to ``fastdds::rtps``
* Make the alias to ``fastrtps::rtps`` in ``include/fastrtps/rtps/common/Locator.h``.